File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RMcpItemUseOnBlockRequest.java
package calebxzhou.rdi.mc.common2.mcp;

public record RMcpItemUseOnBlockRequest(
        String pos,
        String face,
        String itemId,
        Integer fromInventorySlot,
        String hand,
        Integer times,
        boolean dryRun
) {
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RMcpItemUseOnBlockRequest.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RMcpItemUseOnBlockRequest(
    val pos: String,
    val face: String,
    val itemId: String,
    val fromInventorySlot: Int,
    val hand: String,
    val times: Int,
    val dryRun: Boolean
)
